### PR TITLE
chore: documentation for Explorer.Chain.Transaction.History.Historian

### PR DESCRIPTION
**Pull request by akolotov**
## Motivation

Documentation for the module `Explorer.Chain.Transaction.History.Historian` and other has been extended.

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/setup/env-variables/README.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/setup/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/setup/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

## Summary by CodeRabbit

- **New Features**
  - Enhanced documentation and type specifications for various functions in the transaction history and denormalization modules.
  - Introduced a new function for improved configuration handling within the history process module.

- **Bug Fixes**
  - Improved logic for handling transaction statistics and block range determination, ensuring accurate data processing.

- **Documentation**
  - Updated module and function documentation to reflect changes in functionality and clarify usage.

- **Refactor**
  - Revised type definitions for records to provide a more structured format in the historian modules.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/transaction/history/transaction_stats.ex [hunk 1]
```diff
@@ -28,7 +28,26 @@ defmodule Explorer.Chain.Transaction.History.TransactionStats do
...
+    - `options`: Optional keyword list of options used to select the repo for the
+      query.
+
+    ## Returns
+    A list of `Explorer.Chain.Transaction.History.TransactionStats` structs,
+    each representing the transaction statistics for a single day within the
+    specified range.
+  """
   @spec by_date_range(Date.t(), Date.t()) :: [__MODULE__]
+  @spec by_date_range(Date.t(), Date.t(), keyword()) :: [__MODULE__]
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/chain/transaction/history/transaction_stats.ex:50]: @akolotov wdyt?

**Review by vbaranov — COMMENTED** [review]
Please fix `cspell` findings.

**Comment by akolotov** [comment]
> Please fix `cspell` findings.

Done. https://github.com/blockscout/blockscout/pull/11397/commits/8a015a9cc16e4dad3b9b428a6eb07ba16cdd9446

File: apps/explorer/lib/explorer/chain/transaction/history/historian.ex [hunk 2]
```diff
@@ -15,17 +15,59 @@
...
     if num_days == 1 do
       Logger.info("tx/per day chart: records collected #{inspect(records)}")
 
+      # The recourse is finished, and the stats for the current day are set to zero
+      # to avoid presenting incomplete data.
       records = [%{date: date_today(), number_of_transactions: 0, gas_used: 0, total_fee: 0} | records]
-      # base case
       {:ok, records}
     else
+      # Сalculate the date for which the stats are required by subtracting the specified
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/explorer/lib/explorer/chain/transaction/history/historian.ex]: @coderabbitai done in 8a015a9

File: apps/explorer/lib/explorer/chain/transaction/history/transaction_stats.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/explorer/lib/explorer/chain/transaction/history/transaction_stats.ex]: @coderabbitai done in 0a94c4c

[the remark this answers is not in the document]
Comment (baxitfund) [reply apps/explorer/lib/explorer/chain/transaction/history/transaction_stats.ex]: Comfirm

**Review by vbaranov — APPROVED** [review]

**Pull request merged**